downgrade.jl

```diff
@@ -8,6 +8,277 @@ julia_version = length(ARGS) >= 4 ? ARGS[4] : "1.10"
+"""
+function get_local_source_packages(project_file::String)
+    local_pkgs = Set{String}()
+
+    if !isfile(project_file)
+        return local_pkgs
+    end
+
+    project = TOML.parsefile(project_file)
+
+    # Check for [sources] section entries with path keys
+    sources = get(project, "sources", Dict())
+    for (pkg_name, source_info) in sources
+        if source_info isa Dict && haskey(source_info, "path")
+            push!(local_pkgs, pkg_name)
+            @info "Found local source package: $pkg_name (path=$(source_info["path"]))"
+        end
+    end
+
+    return local_pkgs
+end
+
+"""
+    remove_local_packages_from_project(project_file, local_pkgs)
+
+Create a modified version of the Project.toml with local source packages
+removed from [deps], [compat], [extras], and [sources] sections.
+Returns the original content so it can be restored later.
+
+Note: We must also remove from [sources] because Pkg validates that any
+package in [sources] must be in [deps] or [extras].
+"""
+function remove_local_packages_from_project(project_file::String, local_pkgs::Set{String})
+    if isempty(local_pkgs)
+        return nothing  # No modification needed
+    end
+
+    original_content = read(project_file, String)
+    project = TOML.parsefile(project_file)
+    modified = false
+
+    # Remove from [deps]
+    if haskey(project, "deps")
+        for pkg in local_pkgs
+            if haskey(project["deps"], pkg)
+                delete!(project["deps"], pkg)
+                modified = true
+                @info "Temporarily removing $pkg from [deps] for resolution"
+            end
+        end
+    end
+
+    # Remove from [extras]
+    if haskey(project, "extras")
+        for pkg in local_pkgs
+            if haskey(project["extras"], pkg)
+                delete!(project["extras"], pkg)
+                modified = true
+                @info "Temporarily removing $pkg from [extras] for resolution"
+            end
+        end
+    end
+
+    # Remove from [compat]
+    if haskey(project, "compat")
+        for pkg in local_pkgs
+            if haskey(project["compat"], pkg)
+                delete!(project["compat"], pkg)
+                modified = true
+                @info "Temporarily removing $pkg from [compat] for resolution"
+            end
+        end
+    end
+
+    # Remove from [sources] - must do this because Pkg validates that
+    # packages in [sources] must be in [deps] or [extras]
+    if haskey(project, "sources")
+        for pkg in local_pkgs
+            if haskey(project["sources"], pkg)
+                delete!(project["sources"], pkg)
+                modified = true
+                @info "Temporarily removing $pkg from [sources] for resolution"
+            end
+        end
+        # Remove empty [sources] section
+        if isempty(project["sources"])
+            delete!(project, "sources")
+        end
+    end
+
+    if modified
+        open(project_file, "w") do io
```

Comment:
I'm wondering if it's possible/better to write to a temporary project file instead of the actual project file. Will the resolver look at anything besides the project file while creating the manifest?